OpenRA.Mods.Common/Traits/Buildings/PlaceBuildingVariant.cs

```diff
@@ -0,0 +1,28 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Place a different building when PlaceBuilding's ToggleVariantKey hotkey is pressed while the PlaceBuildingOrderGenerator is active.")]
+	public class PlaceBuildingVariantInfo : TraitInfo<PlaceBuildingVariant>, Requires<BuildingInfo>, Requires<BuildableInfo>
+	{
+		[FieldLoader.Require]
+		[ActorReference(typeof(BuildingInfo))]
+		[Desc("Alternate building actor to place.")]
+		public readonly string Actor = null;
```

Comment:
Imho that would be better implemented as list of actors.

Reply:
Ok, done.